java/test/jmri/util/JUnitUtil.java

```diff
@@ -387,5 +387,20 @@ public static void initStartupActionsManager() {
                 apps.StartupActionsManager.class);
     }
 
+    /*
+     * Use reflection to reset the jmri.Application instance
+     */
+    public static void resetApplication() {
+       try {
+          Class<?> c = jmri.Application.class;
+          java.lang.reflect.Field f = c.getDeclaredField("name");
+          f.setAccessible(true);
+          f.set(new jmri.Application(),null);
+       } catch(NoSuchFieldException | IllegalArgumentException | IllegalAccessException x) { 
+         log.error("Failed to reset jmri.Application static field");
```

Comment:
You'll get the stack trace for x by adding x as the final parameter to this logging statement (without any matching {}s).